src/lib/customPropTypes.js

```diff
@@ -170,9 +174,9 @@ export const demand = (requiredProps) => {
     }
 
     // do not require requiredProps if the prop does not exist in props
-    if (!_.has(props, propName)) return
+    if (_.isUndefined(props, propName)) return
 
-    const missingRequired = requiredProps.filter(required => !_.has(props, required))
+    const missingRequired = requiredProps.filter(required => _.isUndefined(props, required))
```

Comment:
Don't demand props when the prop at hand is `undefined`.  React treats `undefined` props as not present.  Previously, `_.has` here was demanding props when the prop implementing `demand()` was undefined.